load/options.go

```diff
@@ -1,79 +1,176 @@
 package load
 
-import "os"
+import (
+	"os"
 
-type Options struct {
-	// ReadFunc reads the config file and returns its content.
-	//
-	// Default: os.ReadFile.
-	ReadFunc ReadFunc
+	"github.com/tableauio/tableau/format"
+	"google.golang.org/protobuf/proto"
+)
+
+// BaseOptions is the common struct for both global-level and messager-level
+// options.
+type BaseOptions struct {
 	// Location represents the collection of time offsets in use in
 	// a geographical area.
 	//
+	// NOTE: only input formats(Excel, CSV, XML, YAML) are supported.
+	//
 	// If the name is "" or "UTC", LoadLocation returns UTC.
 	// If the name is "Local", LoadLocation returns Local.
 	//
 	// Default: "Local".
 	LocationName string
+
 	// IgnoreUnknownFields signifies whether to ignore unknown JSON fields
 	// during parsing.
 	//
-	// Default: false.
-	IgnoreUnknownFields bool
-	// SubdirRewrites rewrites subdir paths (relative to workbook name option
-	// in .proto file).
+	// NOTE: only JSON format is supported.
 	//
 	// Default: nil.
-	SubdirRewrites map[string]string
-	// Paths maps each messager name to a corresponding config file path.
-	// If specified, then the main messager will be parsed from the file
-	// directly, other than the specified load dir.
+	IgnoreUnknownFields *bool

```

Comment:
Is it a good idea to use `*bool` to represent field presence? How about other programming languages, such as C++, C#, and TypeScript?